src/lib/statistics/statistics_objects/abstract_histogram.cpp

```diff
@@ -0,0 +1,1035 @@
+    case PredicateCondition::BetweenExclusive: {
+      Assert(static_cast<bool>(variant_value2), "Between operator needs two values.");
+
+      if (_does_not_contain(PredicateCondition::GreaterThanEquals, *value)) {
+        return true;
+      }
+
+      const auto value2 = static_variant_cast<T>(*variant_value2);
+      if (!value2) {
+        return false;
+      }
+      if (_does_not_contain(PredicateCondition::LessThanEquals, *value2) || value2 < value) {
+        return true;
+      }
+
+      const auto value_bin = _bin_for_value(*value);
+      const auto value2_bin = _bin_for_value(*value2);
+
+      // In an EqualDistinctCountHistogram, if both values fall into the same gap, we can prune the predicate.
+      // We need to have at least two bins to rule out pruning if value < min and value2 > max.
+      if (value_bin == INVALID_BIN_ID && value2_bin == INVALID_BIN_ID && bin_count() > 1ul &&
+          _next_bin_for_value(*value) == _next_bin_for_value(*value2)) {
+        return true;
+      }
+
+      // In an EqualWidthHistogram, if both values fall into a bin that has no elements,
+      // and there are either no bins in between or none of them have any elements, we can also prune the predicate.
+      if (value_bin != INVALID_BIN_ID && value2_bin != INVALID_BIN_ID && bin_height(value_bin) == 0 &&
+          bin_height(value2_bin) == 0) {
+        for (auto current_bin = value_bin + 1; current_bin < value2_bin; current_bin++) {
+          if (bin_height(current_bin) > 0ul) {
+            return false;
+          }
+        }
+        return true;
+      }
+
+      return false;
+    }
+    case PredicateCondition::Like:
+    case PredicateCondition::NotLike:
+      Fail("Predicate (NOT) LIKE is not supported for non-string columns.");
+    default:
+      // Do not prune predicates we cannot (yet) handle.
+      return false;
+  }
+}
+
+template <typename T>
+bool AbstractHistogram<T>::_does_not_contain(const PredicateCondition predicate_condition,
+                                             const AllTypeVariant& variant_value,
+                                             const std::optional<AllTypeVariant>& variant_value2) const {
+  return _general_does_not_contain(predicate_condition, variant_value, variant_value2);
+}
+
+template <>
+bool AbstractHistogram<pmr_string>::_does_not_contain(const PredicateCondition predicate_condition,
+                                                      const AllTypeVariant& variant_value,
+                                                      const std::optional<AllTypeVariant>& variant_value2) const {
+  const auto value = static_variant_cast<pmr_string>(variant_value);
+  if (!value) {
+    return false;
+  }
+
+  switch (predicate_condition) {
+    case PredicateCondition::Like: {
+      if (!LikeMatcher::contains_wildcard(*value)) {
+        return _does_not_contain(PredicateCondition::Equals, *value);
+      }
+
+      // If the pattern starts with a MatchAll, we can not prune it.
```

Comment:
A list of patterns that are (not) handled would be a great start

Reply:
With the Between to >= AND <= rewrite rule, we might not need these estimations anymore. in person.

Reply:
iirc, we decided to kill this.

Reply:
killed